dali/test/python/test_pipeline.py

```diff
@@ -63,6 +63,49 @@ def iter_setup(self):
         t_gpu = a_gpu.at(i)
         assert(np.sum(np.abs(t_cpu - t_gpu)) == 0)
 
+def test_pipeline_separated_exec_setup():
+    batch_size = 128
+    class HybridPipe(Pipeline):
+        def __init__(self, batch_size, num_threads, device_id, num_gpus):
+            super(HybridPipe, self).__init__(batch_size,
+                                             num_threads,
+                                             device_id)
+            self.input = ops.CaffeReader(path = caffe_db_folder, shard_id = device_id, num_shards = num_gpus)
+            self.decode = ops.HostDecoder(device = "cpu", output_type = types.RGB)
+            self.dump_cpu = ops.DumpImage(device = "cpu", suffix = "cpu")
+            self.dump_gpu = ops.DumpImage(device = "gpu", suffix = "gpu")
+
+        def define_graph(self):
+            inputs, labels = self.input(name="Reader")
+            images = self.decode(inputs)
+            images_cpu = self.dump_cpu(images)
+            images_gpu = self.dump_gpu(images.gpu())
+            return (images, images_cpu, images_gpu)
+
+        def iter_setup(self):
+            pass
+
+    pipe = HybridPipe(batch_size=batch_size, num_threads=1, device_id = 0, num_gpus = 1)
+    pipe.set_execution_types(exec_separated=True)
```

Comment:
I think we should keep them as arguments for consistence. As you pointed earlier, and I understand, from a C++ perspective it's not ideal.
The best thing to do is keeping them as separate functions in C++, but wrapping them as ctor argument in the Python API ( (that looks nice thanks to kwargs) through pybinb.